tests/dto/factory/test_integration.py

```diff
@@ -176,3 +176,26 @@ def handler(data: DTOData[User]) -> User:
     with create_test_client(route_handlers=[handler], debug=True) as client:
         response = client.patch("/", json={"age": 41, "read_only": "whoops"})
         assert response.json() == {"name": "John", "age": 41, "read_only": "read-only"}
+
+
+def test_dto_openapi_model_name_collision() -> None:
+    @dataclass
+    class Bar:
+        id: int
+        foo: str
+
+    write_dto = DataclassDTO[Annotated[Bar, DTOConfig(exclude={"id"})]]
+    read_dto = DataclassDTO[Bar]
+
+    @post(dto=write_dto, return_dto=read_dto, signature_namespace={"Bar": Bar})
+    def handler(data: Bar) -> Bar:
+        return data
+
+    with create_test_client(route_handlers=[handler], debug=True) as client:
+        response = client.get("/schema/openapi.json")
+        schemas = list(response.json()["components"]["schemas"].values())
+        assert len(set(schemas)) == 2
```

Comment:
```suggestion
        assert len(schemas) == 2
        assert schemas[0] != schemas[1]
```
This is where the tests are failing, and is due to `schemas` being a `list` of `dict`. Converting that list into a `set` won't work as anything that goes into a `set` needs to be hashable, and `dict`'s are not hashable.

Reply:
Thanks : This resulting in error unhashable type. my bad.

Reply:
tested , and it is passing .